src/GraphQLParser/AST/Values/GraphQLBooleanValue.cs

```diff
@@ -6,15 +6,25 @@ namespace GraphQLParser.AST;
 /// AST node for <see cref="ASTNodeKind.BooleanValue"/>.
 /// </summary>
 [DebuggerDisplay("GraphQLBooleanValue: {Value}")]
-public class GraphQLBooleanValue : GraphQLValue
+public class GraphQLBooleanValue : GraphQLValue, IHasValueNode
 {
+    private readonly bool _value;
+
     /// <inheritdoc/>
     public override ASTNodeKind Kind => ASTNodeKind.BooleanValue;
 
+    /// <summary>
+    /// Creates a new instance with the specified value.
+    /// </summary>
+    public GraphQLBooleanValue(bool value)
+    {
+        _value = value;
+    }
+
     /// <summary>
     /// Boolean value represented as <see cref="ROM"/>.
     /// </summary>
-    public ROM Value { get; set; }
+    public ROM Value => _value ? "true" : "false";
```

Comment:
I think we should add a property to return the underlying boolean.  Saves another comparison in the main project.

Reply:
done